app/src/main/java/org/schabi/newpipe/SponsorBlockApiTask.java

```diff
@@ -0,0 +1,174 @@
+    private static final int timeoutPeriod = 30;
+    private static final String TAG = SponsorBlockApiTask.class.getSimpleName();
+    private static final boolean DEBUG = MainActivity.DEBUG;
+    private OkHttpClient client;
+
+    // api methods
+    public SponsorTimeInfo getVideoSponsorTimes(String url) throws ExecutionException, InterruptedException, JSONException {
+        String videoId = parseIdFromUrl(url);
+        String apiSuffix = "getVideoSponsorTimes?videoID=" + videoId;
+
+        JSONObject obj = execute(apiSuffix).get();
+        JSONArray arrayObj = obj.getJSONArray("sponsorTimes");
+
+        SponsorTimeInfo result = new SponsorTimeInfo();
+
+        for (int i = 0; i < arrayObj.length(); i++) {
+            JSONArray subArrayObj = arrayObj.getJSONArray(i);
+
+            double startTime = subArrayObj.getDouble(0) * 1000;
+            double endTime = subArrayObj.getDouble(1) * 1000;
+
+            TimeFrame timeFrame = new TimeFrame(startTime, endTime);
+
+            result.timeFrames.add(timeFrame);
+        }
+
+        return result;
+    }
+
+    public void postVideoSponsorTimes(String url, double startTime, double endTime, String userId) {
+        if (userId == null) {
+            userId = getRandomUserId();
+        }
+
+        double dStartTime = startTime / 1000.0;
+        double dEndTime = endTime / 1000.0;
+
+        String videoId = parseIdFromUrl(url);
+        String apiSuffix = "postVideoSponsorTimes?videoID=" + videoId + "&startTime=" + dStartTime + "&endTime=" + dEndTime + "&userID=" + userId;
+
+        execute(apiSuffix);
+    }
+
+    // task methods
+    @Override
+    protected JSONObject doInBackground(String... strings) {
+        if (isCancelled() || !isConnected()) return null;
+
+        try {
+            if (client == null) {
+                client = getUnsafeOkHttpClient()
+                        .newBuilder()
+                        .readTimeout(timeoutPeriod, TimeUnit.SECONDS)
+                        .build();
+            }
+
+            Request request = new Request.Builder()
+                    .url(sponsorBlockApiUrl + strings[0])
+                    .build();
+
+            Response response = client.newCall(request).execute();
+            ResponseBody responseBody = response.body();
+
+            return responseBody == null
+                    ? null
+                    : new JSONObject(responseBody.string());
+
+        }
+        catch (Exception ex) {
+            if (DEBUG) Log.w(TAG, Log.getStackTraceString(ex));
+        }
+
+        return null;
+    }
+
+    // helper methods
+    private boolean isConnected() {
+        ConnectivityManager cm = (ConnectivityManager) app.getSystemService(Context.CONNECTIVITY_SERVICE);
+        return cm.getActiveNetworkInfo() != null && cm.getActiveNetworkInfo().isConnected();
+    }
+
+    private OkHttpClient getUnsafeOkHttpClient() throws NoSuchAlgorithmException, KeyManagementException {
```

Comment:
Why are you doing this? https://api.sponsor.ajay.app/ has a valid TLS certificate, so this shouldn't be needed. Please just use the DownloaderImpl class. That way we could e.g. set a proxy in just one file without having to change it everywhere.

Reply:
done, now using DownloaderImpl.getInstance().get(...)